### DISCUSSION TITLE
Problem and Solution: Livewire encountered corrupt data when trying to hydrate the component

### DISCUSSION THREAD

**Discussion by alnutile** [livewire/livewire#4372] [2. Show and tell]
I hit this issue pretty hard and want to share how I resolved it.

This app is months old, integrates with 5 external APIs and takes in and shows data via those APIs, and CSV files so this is nothing new until I hit this one API.

The user would choose a start date and end data and click "Get Preview" then I would hit the API and get a preview.

I tried to move that preview to it's own component in an act of troubleshooting the issue but I still got the "Livewire encountered corrupt data when trying to hydrate the component" error if the user would then choose a new start date and click "Get Preview"

I even did Wireables :) 

Anyways what worked for me was this. I removed "$results" property from the Class attribute and just put it in the `render` area.

So what was this 

```
public $results;

    public function render()
    {
        $dto = FooClient::getPreview(
            $this->guid,
            $this->start_year,
            $this->end_year
        );
        $this->results = new FooResultsWireable($dto);
        return view('livewire.foo-results');
    }
``` 

became this

```
    public function render()
    {
        $dto = FooClient::getPreview(
            $this->guid,
            $this->start_year,
            $this->end_year
        );
        $results = new FooResultsWireable($dto);
        return view('livewire.foo-results', ['results' => $results]);
    }
```

And then it all just worked.  Maybe I got lucky and this is the first time I hit this for some reason but just a heads up to anyone else who hits this issue.

**Comment by joshhanley** [comment]
@alnutile thanks for sharing 🙂

**Reply by teofanis** [reply]
@alnutile @joshhanley I thought to throw this in here in case it helps somebody : 
We found out that the corrupt data was caused by the array keys. Not sure at which stage it happened (probably hydration/dehydration), but when debugging it we noticed that the array keys where changing from int to string type, which was making a different hash and  livewire was throwing this error. We ended up, passing the array through array_values() to recreate the array and avoid the issue.

**Reply by joshhanley** [reply]
@teofanis thanks! Yeah you can thank PHP for that, sometimes weird shit like that happens with array keys. I stumbled across a good article that explains it at one time (but I can't find it 😞), if I do, I'll post it up!

**Comment by pakogn** [comment]
Recently I faced this and it was CloudFlare service.

I would recommend disable Auto Minify, because it removes comments used by livewire to work.

https://developers.cloudflare.com/speed/optimization/content/auto-minify/

Hope this help.
